docs/wiki/The library - Module design.md

```diff
@@ -391,9 +391,12 @@ Within a bicep file, use the following conventions:
   - `Conditional` - The parameter value can be optional or required based on a condition, mostly based on the value provided to other parameters.
   - `Optional` - The parameter value is not mandatory. The module provides a default value for the parameter.
   - `Generated` - The parameter value is generated within the module and should not be specified as input.
-- If parameters map to resource properties, they should not contain the resource's name as it would introduce redundancy. For example, the `name` property of the Key Vault module should be just that and not `keyVaultName`. The rationale is that the consumers know that the name is for the Key Vault if they deploy its module.
-- Further, if a property value allows a single value only, there is no need to introduce a parameter for it. Instead it can be hardcoded into the deployment. For example, the name of a blobServices resource can only be `default`. Hence we can implement its name property directly as `name: 'default'`.
-  > Special case _Diagnostic Settings_: In cases where the resource name can be hardcoded, also the default value for the diagnostic settings name is affected. In those cases, we recommend to introduce an additional variable `'var name = '<theHardcodedValue>'`' (e.g., `var name = 'default'`) to be used both in the main resource's name (e.g., `'name: name'`), as well as the diagnostic settings name: `'name: !empty(diagnosticSettingsName) ? diagnosticSettingsName : '${name}-diagnosticSettings'`'. To make this value obvious, the description of the `'diagnosticSettingsName'` input parameter should be updated to: `'@description('Optional. The name of the diagnostic setting, if deployed. If left empty, it defaults to "<resourceName>-diagnosticSettings".')'`
+- If parameters map to resource properties, they should not contain the resource's name as it would introduce redundancy.
```

Comment:
After the team's vote last week thus is actually not always true

Reply:
Sorry, just formatted the paragraph but kept the same content that was there before the change. Should I then remove this?

Reply:
Not remove, but I guess we have to re-write it. If not in this PR, please in a subsequent one. 
Instead of saying "a parameter name should not contain a resource type's name" I guess it is rather "a parameter name should not artifictially add a resource type's name".

Reply:
Done. However I'd expect more details around this in future PRs, since this is the point which issue #2676 aimed to address